mailbox/store/src/test/java/org/apache/james/mailbox/store/mail/model/AttachmentMapperTest.java

```diff
@@ -68,41 +70,44 @@ void getAttachmentShouldThrowWhenNonReferencedAttachmentId() {
             .isInstanceOf(AttachmentNotFoundException.class);
     }
 
+    @Test
+    void storeAttachmentForOwnerShouldReturnSuppliedInformation() throws Exception {
+        String content = "content";
+        byte[] bytes = "payload".getBytes(StandardCharsets.UTF_8);
+
+        Attachment stored = Mono.from(attachmentMapper.storeAttachmentForOwner(content, new ByteArrayInputStream(bytes), OWNER)).block();
+
+        SoftAssertions.assertSoftly(solftly -> {
+            solftly.assertThat(stored.getSize()).isEqualTo(bytes.length);
+            solftly.assertThat(stored.getType()).isEqualTo(content);
+        });
+    }
+
     @Test
     void getAttachmentShouldReturnTheAttachmentWhenReferenced() throws Exception {
-        //Given
-        Attachment expected = Attachment.builder()
-                .bytes("payload".getBytes(StandardCharsets.UTF_8))
-                .type("content")
-                .build();
-        AttachmentId attachmentId = expected.getAttachmentId();
-        Mono.from(attachmentMapper.storeAttachmentForOwner(expected, OWNER)).block();
-        //When
-        Attachment attachment = attachmentMapper.getAttachment(attachmentId);
-        //Then
-        assertThat(attachment).isEqualTo(expected);
+        String content = "content";
+        byte[] bytes = "payload".getBytes(StandardCharsets.UTF_8);
+
+        Attachment stored = Mono.from(attachmentMapper.storeAttachmentForOwner(content, new ByteArrayInputStream(bytes), OWNER)).block();
+
+        Attachment attachment = attachmentMapper.getAttachment(stored.getAttachmentId());
+
+        SoftAssertions.assertSoftly(solftly -> {
+            solftly.assertThat(attachment.getAttachmentId()).isEqualTo(stored.getAttachmentId());
+            solftly.assertThat(attachment.getSize()).isEqualTo(bytes.length);
+            solftly.assertThat(attachment.getType()).isEqualTo(content);
+        });
     }
 
     @Test
-    void getAttachmentShouldReturnTheAttachmentsWhenMultipleStored() throws Exception {
```

Comment:
where is this test?

Reply:
Distinct AttachmentIds are used, this test is a non sens with the new API.

Removing this test is a good thing, it means we are protected **by design** against what this test was defending us against.